source/fe/mapping_q_generic.cc

```diff
@@ -3737,9 +3770,10 @@ add_quad_support_points(const Triangulation<3,3>::cell_iterator &cell,
                ExcInternalError());
 #endif
 
-      // if face at boundary, then ask boundary object to return intermediate
-      // points on it
-      if (face->at_boundary())
+      // if face at boundary or if it belongs to a manifold, then ask the
+      // boundary/manifold object to return intermediate points on it
+      if (face->at_boundary() ||
+          dynamic_cast<const Boundary<3,3> *>(&face->get_manifold()) == NULL)
```

Comment:
the second check will always return true for all interior faces. The default Manifold description is actually a `FlatManifold`. You can always ask a manifold to give intermediate points, and this should work both in the `Boundary` case, as well as in the `Manifold` or `FlatManifold` (the default) case...